### PR TITLE
Initial Telemetry API docs

### PR DESCRIPTION
**Pull request by howardjohn**
Currently, the docs only have Telemetry for tracing. This adds a common
Telemetry API doc for high level info, shared for all 3 types. It also
adds some info about access logging via Telemetry.

We should likely add a similar page for Metrics, but I did not do it
here.
Most of the docs are moved from existing doc

### CODE REVIEW

**Review by Arhell — COMMENTED** [review]
cc @craigbox

File: content/en/docs/tasks/observability/telemetry/index.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+  - providers:
+    - name: localtrace
+    customTags:
+      foo:
+        literal:
+          value: bar
+    randomSamplingPercentage: 100
+{{< /text >}}
+
+This configuration overrides the default provider from `MeshConfig`, setting the mesh default to be the `"ocaltrace`
```
Comment (douglas-reid) [inline content/en/docs/tasks/observability/telemetry/index.md:98]: we have a`"` here instead of a 'l'

**Review by douglas-reid — COMMENTED** [review]
Looks pretty good. We should fast-follow with some metrics examples too.

**Comment by howardjohn** [comment]
@istio/wg-docs-maintainers ptal, planning to link this in the release blog.

**Review by ericvn — APPROVED** [review]
LGTM. Still need to fix at least one link on an announcement page.

**Comment by ericvn** [comment]
One issue is data/features.yaml comes from the istio/enhancements repo and needs to be changed upstream. https://github.com/istio/enhancements/pull/109 created to update that.

**Pull request merged**